tls/ttls/ttls.c

```diff
@@ -0,0 +1,100 @@
+#include <linux/module.h>
+
+#include "ttls.h"
+
+void *calloc(size_t n, size_t size)
+{
+	return kzalloc(n * size, GFP_ATOMIC);
+}
+
+void free(void *ptr)
+{
+	kfree(ptr);
+}
```

Comment:
Please mark the functions as TODO to analyze all the calls over the whole code and carefully change all of them to appropriate kernel routines (probably SLAB allocations can be used) or exclude at all (if the optimization is possible).